test/cjs-sub-module.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+// This tests that sub-module files using the `.cjs` extension are *not*
+// hookable via a normalized module path. Instead one must use the .cjs
+// extension on the hook arg.
+//
+// E.g., a Hook arg of `cjs-sub-module/foo` will **not** hook
+// `./node_modules/cjs-sub-module/foo.cjs`. This is different compared to `.js`
+// file extension usage. The difference is that Node.js's `require()` treats
+// `.js` and `.cjs` differently.
+// See https://nodejs.org/api/modules.html#file-modules
```

Comment:
I understand the reasoning, but it would make instrumenting modules very complicated. The end user API of `require-in-the-middle` is that the export of whatever targeted module is provided to the hook function. The `(package.json).exports` map complicates this as it can resolve the same export name to multiple extensions. In my view, the hook function shouldn't need to know about that detail. It should just receive the right thing.